src/sqlfluff/core/parser/helpers.py

```diff
@@ -1,30 +1,39 @@
 """Helpers for the parser module."""
 
+from typing import Tuple, TYPE_CHECKING
 
-def frame_msg(msg):
+if TYPE_CHECKING:
+    from .segments import BaseSegment
+
+
+def frame_msg(msg: str) -> str:
     """Frame a message with hashes so that it covers five lines."""
     return "\n###\n#\n# {0}\n#\n###".format(msg)
 
 
-def curtail_string(s, length=20):
+def curtail_string(s: str, length=20) -> str:
     """Trim a string nicely to length."""
     if len(s) > length:
         return s[:length] + "..."
     else:
         return s
 
 
-def join_segments_raw(segments):
+def join_segments_raw(segments: Tuple["BaseSegment", ...]) -> str:

```

Comment:
What does the `...` syntax here do? Haven't come across that before 🤔

Reply:
It makes it a arbitrary length tuple of `BaseSegment` rather than a tuple containing a single element. It effectively tells mypy that we're using tuples as a frozen list.